File: src/backend/schemas/interaction.py
"""상호작용 체크 API 스키마."""

from pydantic import BaseModel, Field

from src.backend.models.interaction import ItemType, Severity


class InteractionItem(BaseModel):
    """상호작용 체크 요청의 단일 아이템."""

    item_type: ItemType
    item_id: int


class InteractionCheckRequest(BaseModel):
    """상호작용 체크 요청.

    최소 2개 아이템을 받아 상호작용을 확인한다.
    """

    items: list[InteractionItem] = Field(..., min_length=2, max_length=20)


class InteractionResult(BaseModel):
    """단일 상호작용 결과."""

    item_a_name: str
    item_b_name: str
    severity: Severity
    description: str | None = None
    mechanism: str | None = None
    recommendation: str | None = None
    source: str
    evidence_level: str | None = None
    ai_explanation: str | None = None
    ai_recommendation: str | None = None


class InteractionCheckResponse(BaseModel):
    """상호작용 체크 응답."""

    total_checked: int
    interactions_found: int
    has_danger: bool
    results: list[InteractionResult]
    disclaimer: str = "이 정보는 참고용이며, 의사/약사의 전문적 판단을 대체하지 않습니다."

File: src/web/src/types/interaction.ts
/** 상호작용 체크 관련 타입 정의 — 백엔드 스키마 1:1 매칭. */

export interface InteractionCheckItem {
  item_type: "drug" | "supplement";
  item_id: number;
}

export interface InteractionResult {
  item_a_name: string;
  item_b_name: string;
  severity: "danger" | "warning" | "caution" | "info";
  description: string | null;
  mechanism: string | null;
  recommendation: string | null;
  source: string;
  evidence_level: string | null;
  ai_explanation: string | null;
  ai_recommendation: string | null;
}

export interface InteractionCheckResponse {
  total_checked: number;
  interactions_found: number;
  has_danger: boolean;
  results: InteractionResult[];
  disclaimer: string;
}
